### PR TITLE
Image operation support

### PR DESCRIPTION
**Pull request by williamkibira**
In this update the following changes have been introduced 

- Switch to conan as the project dependency manager instead of depending on locally installed libraries 
- Removed sub-process commands since zfs will not need direct support image operations 
- Domain `Unix Socket` based communication to issue commands to the daemon 
- ASIO C++ for handling all networking and I/O related operations 
- Moved to using instruction based image composition that should help permit multi-stage image building 
- Shift to CMake Presets to allow for partial development support away from FreeBSD

### CODE REVIEW

File: out/build/unixlike-gcc-debug/CMakeFiles/3.26.1/CMakeDetermineCompilerABI_C.bin
Comment (n0madcoder) [inline out/build/unixlike-gcc-debug/CMakeFiles/3.26.1/CMakeDetermineCompilerABI_C.bin:1]: Are these .bin files required to stay in the repo, or can they be created, or get from elsewhere on each developer's environment?

File: out/build/unixlike-gcc-debug/src/CMakeFiles/daemon.dir/bootstrap.cc.o
Comment (n0madcoder) [inline out/build/unixlike-gcc-debug/src/CMakeFiles/daemon.dir/bootstrap.cc.o:1]: Not sure about including the `out/` directory at all, but again, I prefer asking. Is this required somehow?

**Review by n0madcoder — APPROVED** [review]
The only hint would be to exclude the `out/` directory from the git changes, but that is a non-blocking issue.

LGTM!

File: out/build/unixlike-gcc-debug/CMakeFiles/3.26.1/CMakeDetermineCompilerABI_C.bin
Comment (williamkibira) [reply out/build/unixlike-gcc-debug/CMakeFiles/3.26.1/CMakeDetermineCompilerABI_C.bin:1]: No they are actually not supposed to be in there. forgot to add the .gitignore file

**Pull request merged**